spot_wrapper/wrapper.py

```diff
@@ -363,27 +697,98 @@ def __init__(self, username, password, hostname, robot_name, logger, start_estop
+            self._robot_state_task = AsyncRobotState(
+                self._robot_state_client,
+                self._logger,
+                max(0.0, self._rates.get("robot_state", 0.0)),
+                self._callbacks.get("robot_state", None),
+            )
+            self._robot_metrics_task = AsyncMetrics(
+                self._robot_state_client,
+                self._logger,
+                max(0.0, self._rates.get("metrics", 0.0)),
+                self._callbacks.get("metrics", None),
+            )
+            self._lease_task = AsyncLease(
+                self._lease_client,
+                self._logger,
+                max(0.0, self._rates.get("lease", 0.0)),
+                self._callbacks.get("lease", None),
+            )
+            self._front_image_task = AsyncImageService(
+                self._image_client,
+                self._logger,
+                max(0.0, self._rates.get("front_image", 0.0)),
+                self._callbacks.get("front_image", None),
+                self._front_image_requests,
+            )
+            self._side_image_task = AsyncImageService(
+                self._image_client,
+                self._logger,
+                max(0.0, self._rates.get("side_image", 0.0)),
+                self._callbacks.get("side_image", None),
+                self._side_image_requests,
+            )
+            self._rear_image_task = AsyncImageService(
+                self._image_client,
+                self._logger,
+                max(0.0, self._rates.get("rear_image", 0.0)),
+                self._callbacks.get("rear_image", None),
+                self._rear_image_requests,
+            )
+            self._hand_image_task = AsyncImageService(
+                self._image_client,
+                self._logger,
+                max(0.0, self._rates.get("hand_image", 0.0)),
+                self._callbacks.get("hand_image", None),
+                self._hand_image_requests,
+            )
+
+            self._idle_task = AsyncIdle(
+                self._robot_command_client, self._logger, 10.0, self
+            )
+            self._estop_monitor = AsyncEStopMonitor(
+                self._estop_client, self._logger, 20.0, self
+            )
+            self._world_objects_task = AsyncWorldObjects(
+                self._world_objects_client,
+                self._logger,
+                10.0,
+                self._callbacks.get("world_objects", None),
+            )
 
             self._estop_endpoint = None
+            self._estop_keepalive = None
 
-            self._async_tasks = AsyncTasks([
+            robot_tasks = [
                 self._robot_state_task,
                 self._robot_metrics_task,
                 self._lease_task,
+                self._front_image_task,
                 self._idle_task,
+                self._estop_monitor,
                 self._world_objects_task,
-            ])
+            ]
+
+            if self._point_cloud_client:
```

Comment:
We can use the build int point cloud service, but I would instead recommend using the ROS based one as the robot is quite slow at publishing that much data. Using depth_image_proc is much more performant.

Reply:
I'm not sure I understand what you mean by a ROS based one. We have to retrieve the pointclouds using the SDK interaction before we can pass them to ROS. Perhaps you mean that we should split this out like you did with the image publishing?

Reply:
The package `depth_image_proc` here: http://wiki.ros.org/depth_image_proc. This package takes in an RGB image and a depth image and publishes the point cloud. This means the robot doesn't have to transfer the point cloud data but instead can just publish a depth and rgb image only. This can be added as a launch file which I already did in the `spot_ros2` repo.

Reply:
Ah, I see that we're probably talking about two different things now. The point cloud client isn't used to publish the pointclouds from the cameras, but the one which comes from the VLP16 from the official EAP that BD provides.

Reply:
I think it would be pretty easy to add this parameter but I'm not sure I really see the point of it - so long as the rate is set to negative or no callback is assigned, this just creates a client that isn't used, which doesn't seem like a big deal.

Reply:
Actually rather than creating the parameter, we'd just check if `lidar_points` existed in the callbacks list.